File: backend/src/main/java/com/example/stamp_app/entity/Account.java
package com.example.stamp_app.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Account {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    @NotNull
    private UUID uuid;

    @NotNull
    @Pattern(regexp = "^([a-zA-Z0-9])+([a-zA-Z0-9._-])*@([a-zA-Z0-9_-])+([a-zA-Z0-9._-]+)+$")
    @Column(unique = true, name = "mailAddress")
    private String email;

    @Column
    @NotNull
    private String password;

    @Column
    private String name;

    @Column
    private LocalDateTime createdAt;

    @Column
    private LocalDateTime updatedAt;

    @Column
    private LocalDateTime deletedAt;

    @OneToMany(mappedBy = "account")
    private List<MicroController> microController;

}

File: backend/src/main/java/com/example/stamp_app/entity/Account.kt
package com.example.stamp_app.entity

import jakarta.persistence.*
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import java.time.LocalDateTime
import java.util.*

@Entity
data class Account (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column
    val uuid: @NotNull UUID? = null,

    @Column(unique = true, name = "mailAddress")
    val email: @NotNull @Pattern(regexp = "^([a-zA-Z0-9])+([a-zA-Z0-9._-])*@([a-zA-Z0-9_-])+([a-zA-Z0-9._-]+)+$") String? = null,

    @Column
    val password: @NotNull String? = null,

    @Column
    val name: String? = null,

    @Column
    val createdAt: LocalDateTime? = null,

    @Column
    val updatedAt: LocalDateTime? = null,

    @Column
    val deletedAt: LocalDateTime? = null,

    @OneToMany(mappedBy = "account")
    val microController: List<MicroController>? = null
){
}
